java/yaks-testing-camel-k/src/main/java/dev/yaks/testing/camel/k/CamelKSteps.java

```diff
@@ -89,8 +125,80 @@ private Pod getRunningIntegrationPod(String integration) {
         return null;
     }
 
+	private CustomResourceDefinitionContext getIntegrationCRD() {
+		return new CustomResourceDefinitionContext.Builder()
+				.withName(Integration.CRD_INTEGRATION_NAME)
+				.withGroup(Integration.CRD_GROUP)
+				.withVersion(Integration.CRD_VERSION)
+				.withPlural("integrations")
+				.withScope("Namespaced")
+				.build();
+	}
+
+	private void createIntegration(String name, String source, String dependencies, String traits) {
+		final Integration.Builder integrationBuilder = new Integration.Builder()
+				.name(name)
+				.source(source);
+
+		if(dependencies != null && !dependencies.isEmpty()) {
+			integrationBuilder.dependencies(Arrays.asList(dependencies.split(",")));
+		}
+
+		if (traits != null && !traits.isEmpty()) {
+			final Map<String, Integration.TraitConfig> traitConfigMap = new HashMap<>();
+			for(String t : traits.split(",")){
+				//traitName.key=value
+				if(!validateTraitFormat(t)) {
+					throw new IllegalArgumentException("Trait" + t + "does not match format traitName.key=value");
+				}
+				final String[] trait = t.split("\\.",2);
+				final String[] traitConfig = trait[1].split("=", 2);
+				if(traitConfigMap.containsKey(trait[0])) {
+					traitConfigMap.get(trait[0]).add(traitConfig[0], traitConfig[1]);
+				} else {
+					traitConfigMap.put(trait[0],  new Integration.TraitConfig(traitConfig[0], traitConfig[1]));
+				}
+			}
+			integrationBuilder.traits(traitConfigMap);
+		}
+
+		final Integration i = integrationBuilder.build();
+
+		final CustomResourceDefinitionContext crdContext = getIntegrationCRD();
+
+		try {
+			Map<String, Object> result = client().customResource(crdContext).createOrReplace(namespace(), obj.writeValueAsString(i));
+			if(result.get("message") != null) {
+				throw new IllegalStateException(result.get("message").toString());
+			}
+		} catch (IOException e) {
+			throw new IllegalStateException("Can't create Integration JSON object", e);
+		}
+	}
+
+	private boolean validateTraitFormat(String trait) {
+		String patternString = "[A-Za-z-0-9]+\\.[A-Za-z-0-9]+=[A-Za-z-0-9]+";
+
+		Pattern pattern = Pattern.compile(patternString);
+
+		Matcher matcher = pattern.matcher(trait);
+		return matcher.matches();
+	}
+
+
+
     private String namespace() {
-        return System.getenv("NAMESPACE");
+        String result = System.getenv("NAMESPACE");
+		final File namespace = new File("/var/run/secrets/kubernetes.io/serviceaccount/namespace");
```

Comment:
Can we make this work for non-unix systems (like Windows), too?

Reply:
The test should be running in the OCP container. The container should be always unix-like system.